### ISSUE TITLE
Matrix bridging documentation issue

### ISSUE THREAD

**Issue by axelsimon** [zulip/python-zulip-api#545]
Hi,
I found some difficulties and errors following the guide at https://zulip.com/integrations/doc/matrix.

In "Installing the bridge software", at step 3:
> Install the Matrix bridge software in your virtualenv, by running:
> `pip install -r zulip/integrations/matrix/requirements.txt`

That won't work, as the `matrix` directory is actually called `bridge_with_matrix` (cf. [here](https://github.com/zulip/python-zulip-api/tree/master/zulip/integrations/bridge_with_matrix)).

The documentation needs to be updated to reflect this :)

In "Configure the Bridge", step 2 is just "Subscribe the bot to the Zulip stream that will contain the mirror." which isn't as helpful as it could be, as the link to more help is to a page about adding *users*, not *bots*, to a stream. It works the same for bots, but they a/ there is no mention of that and b/ bots are not listed in the list of users that is shown. To actually subscribe your bot, you need to enter the beginning of the email address you gave them in step 1.

Step 3 says: 
> Inside the virtualenv you created above, run
> `python matrix_bridge.py --write-sample-config matrix_bridge.conf --from-zuliprc <path/to/zuliprc>`

That won't work, as `matrix_bridge.py` doesn't exist in the the root of the virtualenv, it's in `./zulip/integrations/bridge_with_matrix/matrix_bridge.py`, and so the proper command is:
> `python zulip/integrations/bridge_with_matrix/matrix_bridge.py --write-sample-config matrix_bridge.conf --from-zuliprc <path/to/zuliprc>`

Step 5 is a bit vague. It doesn't tell you very clearly that on the matrix side, the room_id is where your Zulip stream will be bridged and that it must *already exist*, for instance.

Step 6: same problem as with step 3: `python matrix_bridge.py -c matrix_bridge.conf` won't work, it needs to be:
`python zulip/integrations/bridge_with_matrix/matrix_bridge.py -c matrix_bridge.conf`

There is more to report (such as the fact the Zulip bridge bot repeats every message i send from Zulip *in Zulip*. That's just not going to be usable…), but hope this helps.

**Comment by timabbott** [comment]
@axelsimon thanks for the report; @orientor just fixed the documentation; can you also report the other issues you saw?

**Comment by timabbott** [comment]
@orientor @rht I'd encourage you to test the bridge somewhat extensively against a real matrix server to QA for additional issues here, since it's clear from this issue that the Matrix bridge hadn't received enough testing

**Comment by orientor** [comment]
@timabbott Sure, I will start testing it.

**Comment by axelsimon** [comment]
> @axelsimon thanks for the report; @orientor just fixed the documentation; can you also report the other issues you saw?

Hi @timabbott, thanks for the reply. Is the fixed documentation still in staging and not yet published?
Because **Configure the Bridge** section 3 and 5 still have the mistake i reported, for instance.

Would you like me to report the other things in this issue or in a new issue?

**Comment by rht** [comment]
> @orientor @rht I'd encourage you to test the bridge somewhat extensively against a real matrix server to QA for additional issues here, since it's clear from this issue that the Matrix bridge hadn't received enough testing

FWIW, I already did the QA with a real Matrix server the first time I made the PR, back when the IRC bridge wasn't working. It's just that I missed the integration name rename review.

**Comment by rht** [comment]
> > @axelsimon thanks for the report; @orientor just fixed the documentation; can you also report the other issues you saw?
> 
> Hi @timabbott, thanks for the reply. Is the fixed documentation still in staging and not yet published?
> Because **Configure the Bridge** section 3 and 5 still have the mistake i reported, for instance.
> 
> Would you like me to report the other things in this issue or in a new issue?

@axelsimon indeed, the README (at https://github.com/zulip/python-zulip-api/blob/8fc5b777db5e1fa879d5b61b88ff80d4bc7110ed/zulip/integrations/bridge_with_matrix/README.md) hasn't been updated to use the path starting from the repo's root. @orientor only updated the documentation of other files. We are still debating whether it is clearer instead to tell the user to `cd` to the appropriate sub-directory.

**Comment by timabbott** [comment]
@rht @orientor does one of you have time to fix the remaining errors here?

@axelsimon I think new issues might be better; this thread is already somewhat cluttered.

**Comment by rht** [comment]
@timabbott it is currently blocked waiting for your reply to https://github.com/zulip/zulip/pull/14249#issuecomment-601992847.
